### PR TITLE
Fixed issue #18316: User role cannot be deleted with action button

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Comment by olleharstedt** [comment]
Why are all checks cancelled? No CI was run?

**Comment by Shnoulle** [comment]
Deleting a role MUST be CRSF controlled, must be a POST action.

ALL db update must be CRSF controlled then must be POST action.

File: application/controllers/UserRoleController.php [hunk 1]
```diff
@@ -250,8 +250,6 @@ public function actionRunExport($ptid)
      */
     public function actionDelete()
     {
-        $this->requirePostRequest();
```
Comment (Shnoulle) [inline application/controllers/UserRoleController.php:253]: Disabling CRSF control is a security issue.

**Review by Shnoulle — CHANGES_REQUESTED** [review]

File: application/controllers/UserRoleController.php (hunk 1 above)
Comment (olleharstedt) [reply application/controllers/UserRoleController.php:253]: Yes, agree, this is not OK.

Comment (Shnoulle) [reply application/controllers/UserRoleController.php:253]: Oh yes : https://github.com/LimeSurvey/LimeSurvey/blob/62d4b30c3ec38c478551c6e35054e8a85cf38ce0/application/controllers/UserRoleController.php#L257

Right , we have a getPost here :)

resolved

**Comment by olleharstedt** [comment]
Tested by me. Thanks!

**Pull request merged**